dist/main.d.ts

```diff
@@ -0,0 +1,16 @@
+import './index.d.ts';
```

Comment:
nit: I don't think we should have "index" and "main" as those both denote main files. We could move the existing `index.d.ts` to another filename though.

Reply:
moved it to `types.d.ts`